tests/child_processes_test.py

```diff
@@ -139,9 +138,9 @@ def test_processes_dont_receive_term_on_exit_if_no_setsid():
 def test_fails_nonzero_with_bad_exec(args):
     """If dumb-init can't exec as requested, it should exit nonzero."""
     proc = Popen(('dumb-init',) + args, stderr=PIPE)
-    proc.wait()
+    _, stderr = proc.communicate()
     assert proc.returncode != 0
     assert (
         b'[dumb-init] /doesnotexist: No such file or directory\n'
-        in proc.stderr
+        in stderr
```

Comment:
Hypothesis is that `proc.wait` deadlocked when `dumb-init` was trying to write with a full buffer, but not really clear how that could happen since it writes so little.

```
21:49:56 ============================= test session starts ==============================
21:49:56 platform linux -- Python 3.4.2, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
21:49:56 rootdir: /ephemeral/jenkins_prod_slave/workspace/mirrors-Yelp-dumb-init, inifile: pytest.ini
21:49:56 plugins: timeout-1.0.0
21:49:57 collected 173 items
21:49:57 
21:50:02 tests/child_processes_test.py ......................F.
21:50:03 tests/cli_test.py ......................................................................
21:50:03 tests/exit_status_test.py ................................................
21:50:04 tests/proxies_signals_test.py ..........................
21:50:05 tests/shell_background_test.py ....
21:50:05 tests/tty_test.py .
21:50:05 
21:50:05 =================================== FAILURES ===================================
21:50:05 _________________ test_fails_nonzero_with_bad_exec[0-0-args2] __________________
21:50:05 
21:50:05 args = ('-c', '/doesnotexist')
21:50:05 
21:50:05     @pytest.mark.parametrize('args', [
21:50:05         ('/doesnotexist',),
21:50:05         ('--', '/doesnotexist'),
21:50:05         ('-c', '/doesnotexist'),
21:50:05         ('--single-child', '--', '/doesnotexist'),
21:50:05     ])
21:50:05     @pytest.mark.usefixtures('both_debug_modes', 'both_setsid_modes')
21:50:05     def test_fails_nonzero_with_bad_exec(args):
21:50:05         """If dumb-init can't exec as requested, it should exit nonzero."""
21:50:05         proc = Popen(('dumb-init',) + args, stderr=PIPE)
21:50:05 >       proc.wait()
21:50:05 
21:50:05 tests/child_processes_test.py:142: 
21:50:05 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
21:50:05 /usr/lib/python3.4/subprocess.py:1565: in wait
21:50:05     (pid, sts) = self._try_wait(0)
21:50:05 /usr/lib/python3.4/subprocess.py:1513: in _try_wait
21:50:05     (pid, sts) = _eintr_retry_call(os.waitpid, self.pid, wait_flags)
21:50:05 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
21:50:05 
21:50:05 func = <built-in function waitpid>, args = (25548, 0)
21:50:05 
21:50:05     def _eintr_retry_call(func, *args):
21:50:05         while True:
21:50:05             try:
21:50:05 >               return func(*args)
21:50:05 E               Failed: Timeout >5s
21:50:05 
21:50:05 /usr/lib/python3.4/subprocess.py:491: Failed
```